core/src/blockchain/headerchain.rs

```diff
@@ -267,20 +272,35 @@ impl HeaderChain {
         block_details
     }
 
+    /// Compare the number and the view of current block with these of best block
+    fn compare_number_and_view_to_best(&self, new_header: &HeaderView<'_>) -> bool {
+        let best_proposal_block_hash = self.best_header_hash();
+        let best_proposal_block_detail = self.block_details(&best_proposal_block_hash).expect("Best proposal block always exists");
+
+        let compare_number = new_header.number() > best_proposal_block_detail.number;
+        let compare_view = if new_header.view() == None {
+            true
+        } else {
+            new_header.view() > best_proposal_block_detail.view
+        };
+        
+        compare_number && compare_view
```

Comment:
AFAIK, the previous score was designed to pick one block ordered by score so the score formula for Tendermint could be seen arbitrary. However, this number and view compare logic changes the previous ordering.
As I mentioned before, (new_header.number(), best_proposal_block_detail.view()) > (best_proposal_block_detail.number, new_header.view()) seems appropriate because the it follows the previous implication "Pick the higher block number one and pick the lower number one only when the block numbers are equal" but this comparing implies "Pick the one who has higher block number and view".
It seems the test failures are not related directly to this comparing but It's just a comment on consistency.